File: Python/Medium/Stack/853__Car_Fleet.py
def carFleet(target: int, position: list[int], speed: list[int]) -> int:
    cars = sorted(zip(position, speed), reverse=True)
    stack = []

    for pos, spd in cars:
        time = (target - pos) / spd

        if not stack or time > stack[-1]:
            stack.append(time)
    return len(stack)


target = 12
position = [10, 8, 0, 5, 3]
speed = [2, 4, 1, 1, 3]
print(carFleet(target, position, speed))

File: PHP/Medium/Stack/853__Car_Fleet.php
<?php
// https://leetcode.com/problems/car-fleet/
function carFleet($target, $position, $speed) {
    $cars = array_map(null, $position, $speed);
    usort($cars, function ($a, $b) {return $b[0] <=> $a[0];});

    $stack = [];
    foreach ($cars as $car) {
        $pos = $car[0];
        $spd = $car[1];
        $time = ($target - $pos) / $spd;

        if (empty($stack) || $time > end($stack)) {
            $stack[] = $time;
        }
    }
    return count($stack);

}